scripts/build_android.sh

```diff
@@ -19,58 +19,77 @@
 # (3) The toolchain and the build target platform can be specified with the
 #     cmake arguments below. For more details, check out android-cmake's doc.
 
+set -e
+
 CAFFE2_ROOT="$( cd "$(dirname "$0")"/.. ; pwd -P)"
-echo "Caffe2 codebase root is: $CAFFE2_ROOT"
 
 if [ -z "$ANDROID_NDK" ]; then
-    echo "Did you set ANDROID_NDK variable?"
-    exit 1
+  echo "ANDROID_NDK not set; please set it to the Android NDK directory"
+  exit 1
 fi
 
-if [ -d "$ANDROID_NDK" ]; then
-    echo "Using Android ndk at $ANDROID_NDK"
-else
-    echo "Cannot find ndk: did you install it under $ANDROID_NDK?"
-    exit 1
+if [ ! -d "$ANDROID_NDK" ]; then
+  echo "ANDROID_NDK not a directory; did you install it under $ANDROID_NDK?"
+  exit 1
 fi
-# We are going to build the target into build_android.
-BUILD_ROOT=$CAFFE2_ROOT/build_android
-mkdir -p $BUILD_ROOT
-echo "Build Caffe2 Android into: $BUILD_ROOT"
+
+echo "Using Android NDK at $ANDROID_NDK"
 
 # Build protobuf from third_party so we have a host protoc binary.
 echo "Building protoc"
-$CAFFE2_ROOT/scripts/build_host_protoc.sh || exit 1
+$CAFFE2_ROOT/scripts/build_host_protoc.sh
 
-# Now, actually build the android target.
-echo "Building caffe2"
+# Now, actually build the Android target.
+BUILD_ROOT=$CAFFE2_ROOT/build_android
+mkdir -p $BUILD_ROOT
 cd $BUILD_ROOT
 
+CMAKE_ARGS=()
```

Comment:
Interesting, this line is causing issues here:

https://travis-ci.org/caffe2/caffe2/jobs/302897319#L2002

But it should be correct...